File: sample/venomtyme.go
package main

import "fmt"

/*
  Once upon a time...
*/
type Cat struct {
  Location   string
  BirthDate  int
  CurrentDate  int
  Catlikes []string
}

func (v *Cat) Age() int {
  return v.calcAge()
}

func (v *Cat) calcAge() int {
  return v.CurrentDate - v.BirthDate
}

// ...there was a cat named Katlyn

func main() {
  catppuccin := &Cat{
    Location:   "Cataliputra",
    BirthDate:  2016,
    CurrentDate:  2022,
    Catlikes: []string{"Tasty Food", "Long Naps"},
  }

  fmt.Println(catppuccin.Age())
}

File: sample/venomtyme.js
/*
 * Once upon a time...
 */
class Cat {
  constructor(props) {
    this.location = props.location;
    this.birthDate = props.birthDate;
    this.currentDate = props.currentDate;
    this.catlikes = props.catlikes;
  }

  get age() {
    return this.calcAge();
  }

  calcAge() {
    return this.currentDate - this.birthDate;
  }
}

// ...there was a cat named Katlyn
const Catppuccin = new Cat({
  location: "Cataliputra",
  birthDate: 2016,
  currentDate: 2022,
  catlikes: ["Tasty Food", "Long Naps"],
});
